tests/test_core.py

```diff
@@ -1,139 +1,143 @@
+    wrapper_env.reward_range = (-1.0, 1.0)
+    assert env.reward_range != wrapper_env.reward_range
+
+    assert env.spec == wrapper_env.spec
+
+    env.observation_space = Box(0, 1)
+    env.action_space = Box(0, 1)
+    assert env.observation_space == wrapper_env.observation_space
+    assert env.action_space == wrapper_env.action_space
+    wrapper_env.observation_space = Box(1, 2)
+    wrapper_env.action_space = Box(1, 2)
+    assert env.observation_space != wrapper_env.observation_space
+    assert env.action_space != wrapper_env.action_space
+
+    wrapper_env.np_random, _ = seeding.np_random()
+    assert (
+        env._np_random is env.np_random is wrapper_env.np_random
+    )  # ignore: reportPrivateUsage
+    assert 0 <= wrapper_env.np_random.uniform() <= 1
+    with pytest.raises(
+        AttributeError,
+        match=re.escape(
+            "Can't access `_np_random` of a wrapper, use `self.unwrapped._np_random` or `self.np_random`."
+        ),
+    ):
+        print(wrapper_env.access_hidden_np_random())
 
-    def reset(self):
-        super().reset()
-        return 0
 
-    def step(self, action):
-        return 0, 0, False, {}
+class ExampleRewardWrapper(RewardWrapper):
+    def reward(self, reward: SupportsFloat) -> SupportsFloat:
+        return 1
 
 
-class NewPropertyWrapper(core.Wrapper):
-    def __init__(
-        self,
-        env,
-        observation_space=None,
-        action_space=None,
-        reward_range=None,
-        metadata=None,
-    ):
-        super().__init__(env)
-        if observation_space is not None:
-            # Only set the observation space if not None to test property forwarding
-            self.observation_space = observation_space
-        if action_space is not None:
-            self.action_space = action_space
-        if reward_range is not None:
-            self.reward_range = reward_range
-        if metadata is not None:
-            self.metadata = metadata
-
-
-def test_env_instantiation():
-    # This looks like a pretty trivial, but given our usage of
-    # __new__, it's worth having.
-    env = ArgumentEnv("arg")
-    assert env.arg == "arg"
-    assert env.calls == 1
-
-
-properties = [
-    {
-        "observation_space": spaces.Box(
-            low=0.0, high=1.0, shape=(64, 64, 3), dtype=np.float32
-        )
-    },
-    {"action_space": spaces.Discrete(2)},
-    {"reward_range": (-1.0, 1.0)},
-    {"metadata": {"render_modes": ["human", "rgb_array_list"]}},
-    {
-        "observation_space": spaces.Box(
-            low=0.0, high=1.0, shape=(64, 64, 3), dtype=np.float32
-        ),
-        "action_space": spaces.Discrete(2),
-    },
-]
+class ExampleObservationWrapper(ObservationWrapper):
+    def observation(self, observation: ObsType) -> ObsType:
+        return np.array([1])
 
 
-@pytest.mark.parametrize("class_", [UnittestEnv, UnknownSpacesEnv])
-@pytest.mark.parametrize("props", properties)
-def test_wrapper_property_forwarding(class_, props):
```

Comment:
I'm still a bit hesitant about removing the existing tests.

Are these tests redundant now? Do they fail with the new changes? (if so, that's a potential problem) Are they 100% replaced with the new tests?

I'd lean towards keeping them in, and just adding new tests instead. And if some tests turn out to be unnecessary, we can remove them separately with no other changes.